fuzzing/test_case_assertion_provider.go

```diff
@@ -73,7 +73,7 @@ func (t *AssertionTestCaseProvider) checkAssertionFailures(callSequence calls.Ca
 	// have a panic code.
 	lastExecutionResult := lastCall.ChainReference.MessageResults().ExecutionResult
 	panicCode := abiutils.GetSolidityPanicCode(lastExecutionResult.Err, lastExecutionResult.ReturnData, true)
-	encounteredAssertionFailure := panicCode != nil && panicCode.Uint64() == abiutils.PanicCodeAssertFailed
+	encounteredAssertionFailure := panicCode != nil && utils.IsPanicCodeIncluded(byte(panicCode.Uint64()), t.fuzzer.config.Fuzzing.Testing.AssertionTesting.AssertOptions) && utils.HasEncounteredAssertionFailure(panicCode)
```

Comment:
So this is the right idea but I think we can remove the dependency with `assertion_utils` by storing which panic codes are enabled/disabled directly in the provider. I would look at the `attachAssertionTestCaseProvider` function and see how you can instantiate some structure that would hold which panic codes are enabled. Note that the `attachAssertionTestCaseProvider` takes in a `Fuzzer` object so you have the underlying `Fuzzing.Config` at your disposal. My initial thoughts are to create a mapping (maybe `map[byte]bool` ? but not 100% sure) that you can simply query to see if some `panicCode` is enabled or not

Reply:
So the question is how do you take the `AssertionModeConfig` object, store it in a meaningful way in the `AssertionTestCaseProvider` object, instantiate that object correctly in `attachAssertionTestCaseProvider` and use it accoredingly in `checkAssertionFailures`

Reply:
If you're switching to individual config options you'll need to reference, you can create this mapping in the test case provider on fuzzer start.

Otherwise you can just make it a switch/if-else statement. That may actually be computationally faster than a dictionary lookup given the small number of items and that it's a single byte compare.